PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/console/SparkScalaLivyConsoleRunProfileState.kt

```diff
@@ -44,25 +47,24 @@ class SparkScalaLivyConsoleRunProfileState(private val consoleBuilder: SparkScal
     """.trimIndent()
 
     override fun execute(executor: Executor, runner: ProgramRunner<*>): ExecutionResult? {
-        try {
-            session.create()
-                    .flatMap { it.awaitReady() }
-                    .toBlocking()
-                    .single()
-        } catch (ex: Exception) {
-            throw ExecutionException("Can't start Livy interactive session: ${ex.message}")
-        }
-
         val console = consoleBuilder.console
-        (console as? ScalaLanguageConsole)?.apply {
-            // Customize the Spark Livy interactive console
-            prompt = "\nSpark>"
-        }
 
         val livySessionProcessHandler = SparkLivySessionProcessHandler(SparkLivySessionProcess(session))
 
         console.attachToProcess(livySessionProcessHandler)
-        livySessionProcessHandler.execute(postStartCodes)
+
+        session.create()
+                .subscribeOn(Schedulers.io())
```

Comment:
Could you try to replace the scheduler with `IdeaSchedulers.processBarVisibleAsync()`, which can show the progress in the status bar?

Reply:
Opened issue #2465 to track this